### PR TITLE
[FlexAttention] Remove retriction on QK headdim > V headdim

### PR DESCRIPTION
**Pull request by drisspg**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #135654

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang

### CODE REVIEW

**Comment by drisspg** [comment]
## Standalone repro causing seg fault
```Python
import torch
from torch.nn.attention.flex_attention import flex_attention

B = 4
H = 8
S = 2048
D = 32
V_D=64
dtype = torch.bfloat16

query = torch.randn(B, H, S, D, dtype=dtype, device="cuda", requires_grad=True)
key = torch.randn(B, H, S, D, dtype=dtype, device="cuda", requires_grad=True)
value = torch.randn(B, H, S, V_D, dtype=dtype, device="cuda", requires_grad=True)


def id(score, b, h, q_idx, kv_idx):
    return score

flex = torch.compile(flex_attention, fullgraph=True)
out = flex(query, key, value, score_mod=id)
grad = torch.randn((B, H, S, V_D), dtype=dtype, device="cuda")
out.backward(grad)
print(query.grad.shape)
print(value.grad.shape)
```
Happening in bwds during in triton mm scheduler
```Python
 thread #50, name = 'pt_autograd_0', stop reason = signal SIGSEGV: address not mapped to object (fault address: 0x10)
    frame #0: 0x00007ffcc94812be libtriton.so`scheduleRemainingToLastStage(forOp=ForOp @ 0x00007fff1ebcffc8, schedule=0x00007fff1ebd0350, afterPrologue=<unavailable>, numStages=2) at MatmulLoopPipeline.cpp:893:9
(lldb) bt
* thread #50, name = 'pt_autograd_0', stop reason = signal SIGSEGV: address not mapped to object (fault address: 0x10)
  * frame #0: 0x00007ffcc94812be libtriton.so`scheduleRemainingToLastStage(forOp=ForOp @ 0x00007fff1ebcffc8, schedule=0x00007fff1ebd0350, afterPrologue=<unavailable>, numStages=2) at MatmulLoopPipeline.cpp:893:9
    frame #1: 0x00007ffcc948f030 libtriton.so`mlir::triton::preProcessLoopAndGetSchedule(forOp=0x00007fff1ebd0730, numStages=2, options=0x00007fff1ebd07f0) at MatmulLoopPipeline.cpp:1230:31
    frame #2: 0x00007ffcc94a8103 libtriton.so`mlir::triton::gpu::PipelinePass::runOnOperation() [inlined] pipelineLoop(numStages=2, forOp=ForOp @ 0x00007fff1ebd0730) at SoftwarePipeliner.cpp:79:47
    frame #3: 0x00007ffcc94a8058 libtriton.so`mlir::triton::gpu::PipelinePass::runOnOperation(this=0x00007ffcbc0acc20) at SoftwarePipeliner.cpp:125:36
    frame #4: 0x00007ffcc9a52f2c libtriton.so`mlir::detail::OpToOpPassAdaptor::run(mlir::Pass*, mlir::Operation*, mlir::AnalysisManager, bool, unsigned int) + 700
    frame #5: 0x00007ffcc9a538a2 libtriton.so`mlir::detail::OpToOpPassAdaptor::runPipeline(mlir::OpPassManager&, mlir::Operation*, mlir::AnalysisManager, bool, unsigned int, mlir::PassInstrumentor*, mlir::PassInstrumentation::PipelineParentInfo const*) + 354
    ```

**Review by yanboliang — APPROVED** [review]
